src/App.js

```diff
@@ -14,11 +15,139 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    goods: goodsFromServer,
+    visibleList: false,
+    visibleButton: 'visible',
+    initial: goodsFromServer,
+    selectedLength: 1,
+  }
+
+  showGoods = (event) => {
+    this.setState(state => ({
+      visibleList: !state.visibleList,
+      visibleButton: 'hidden',
+    }));
+  }
+
+  reverseList = () => {
+    this.setState(state => ({
+      goods: [...state.goods].reverse(),
+    }));
+  }
+
+  selectChange = (event) => {
+    const { value } = event.target;
+
+    this.setState(state => ({
+      selectedLength: value,
+      goods: state.initial.filter(item => item.length >= +value),
+    }));
+  }
+
+  sortByAlphabet = () => {
+    this.setState(state => ({
+      goods: [...state.goods].sort(),
+    }));
+  }
+
+  backToInitial = () => {
+    this.setState(state => ({
+      goods: state.initial,
+      selectedLength: 1,
+    }));
+  }
+
+  sortByLength = () => {
+    this.setState(state => ({
+      goods: [...state.goods].sort((a, b) => a.length - b.length),
+    }));
+  }
+
+  render() {
+    const { visibleList, goods, visibleButton } = this.state;
+    const selecting = new Array(10).fill(0);
```

Comment:
create this array outside component, in that case, you create it only once, but in your version, you create it on every render, and you can fill it with indexes as elements and use them as keys